### ISSUE TITLE
make eggs= optional

### ISSUE THREAD

**Issue by djay** [nazrulworld/collective.recipe.vscode#8]
It would be nice if you can add the recipe to ```~/.buildout/default.cfg``` and it will find all eggs without having to know anything about the buildout. This would mean the only thing needed to get vscode support is run a buildout without changing it.

I think this should be possible as I did something similar in collective.hostout. If the recipe installs itself as dependent on all other parts of the buildout it will run last. At that point all resolution is done. Buildout compiles a signature of each recipe based on what eggs it's dependent on. It should be possible to hook into that and read that list of eggs.

**Comment by nazrulworld** [comment]
This sounds a very good idea.

**Comment by djay** [comment]
I did a test. In principle it works but the idea of having it .buildout/default.cfg doesn't really work because there isn't a way to ensure the part is in ```buildout:parts```. I think the only way to do that is to make it an extension rather than a recipe.

**Comment by djay** [comment]
@nazrulworld There is a WIP PR in #9 but I think an exstention would be nice

**Comment by djay** [comment]
@nazrulworld actually I think an extension won't work either. The problem is that if buildout has

```
parts = blah
```

then default.cfg with

```
parts += vscode
```

will never be included. The same applies to extensions.

**Comment by djay** [comment]
@nazrulworld I worked it 
```
    Modify/Add ``~/.buildout/default.cfg``::

        [buildout]
        _dummy = ${vscode:recipe}
        eggs =

        [vscode]
        recipe = collective.recipe.vscode
        eggs = ${buildout:eggs} ${test:eggs} ${instance:eggs}

        [test]
        eggs=

        [instance]
        eggs=
```

**Comment by djay** [comment]
@nazrulworld actually with https://github.com/nazrulworld/collective.recipe.vscode/pull/9 then things are even simpler.
I think that PR is ready to go. maybe could add more tests to get coverage up?

**Comment by djay** [comment]
fixed in #9
